second.c

```diff
@@ -0,0 +1,215 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILENAME "users.txt"
+
+typedef struct
+{
+    int id;
+    char fname[100];
+    char lname[100];
+    int age;
+} User;
+
+void initializeFile()
+{
+    FILE *file = fopen(FILENAME, "r");
+    if (file == NULL)
+    {
+        file = fopen(FILENAME, "a");
+        printf("File has been created. \n");
+    }
+    fclose(file);
+}
+
+void addUser()
+{
+    User user;
+    int id, found = 0;
+
+    // first opening in read mode to check if the id is unique or not
+    FILE *file = fopen(FILENAME, "r");
+
+    printf("Enter User ID: ");
+    scanf("%d", &id);
+
+    while (fscanf(file, "%d %s %s %d", &user.id, user.fname, user.lname, &user.age) != EOF)
+    {
+        if (user.id == id)
+        {
+            found = 1;
+            printf("User ID already exists. Please enter a new one.\n");
+            break;
+        }
+    }
+
+    if (found)
+    {
+        fclose(file);
+        return;
+    }
+    fclose(file);
+
+    // if new user id then append the user
+    file = fopen(FILENAME, "a");
+    if (file == NULL)
+    {
+        printf("Error: Could not open file for writing.\n");
+        return;
+    }
+
+    user.id = id;
+    getchar();
+    printf("Enter First Name: ");
+    fgets(user.fname, 100, stdin);
+    user.fname[strcspn(user.fname, "\n")] = '\0';
+
+    printf("Enter Last Name: ");
+    fgets(user.lname, 100, stdin);
+    user.lname[strcspn(user.lname, "\n")] = '\0';
+
+    printf("Enter Age: ");
+    scanf("%d", &user.age);

```

Comment:
what if someone enter negative age?

Reply:
I have added age validation to ensure negative values aren't accepted. The program prompts the user to re-enter a valid age until a non-negative value is provided, maintaining data integrity.